### PR TITLE
Merge from upstream to include GCC recipe fix

### PR DESCRIPTION
**Pull request by greenc-FNAL**
- **omb-micro-benchmarks: add v7.5.1 (#500)**
- **Require newer version of llvm-openmp for xgboost and py_torch on macOS (#619)**
- **Update OpenTurbine package to add cuDSS variant and to update ADI/ROSCO variants (#420)**
- **force CMAKE_INSTALL_RPATH to prefix/lib/root for ROOT (#436)**
- **Add conflict for package scotch with Intel oneAPI compilers (#195)**
- **Py_xcdat:  new package (#374)**
- **Enable support for C/C++ standards (#366)**
- **intel-oneapi: enable determining openmp runtime (#468)**
- **ci: bump SPACK_CHECKOUT_VERSION (#645)**
- **llvm: update description for flang variant (#636)**
- **bear: add 3.1.4, 3.1.5, 3.1.6 (#635)**
- **arborx: version 2.0.1 (#657)**
- **add 'just' Spack package (#644)**
- **kokkos ecosystem: release 4.6.02 (#633)**
- **py-beautifulsoup4: add v4.13.4 (#631)**
- **GDAL: add v3.11.1-3, fix +arrow build (#561)**
- **angsd: install misc/NGSadmix binary into the bin directory. (#639)**
- **py-beniget: add v0.4.2.post1 (#632)**
- **py-babel: add v2.17.0 (#630)**
- **pkgconf: add 2.5.1 (#628)**
- **py-attrs: add v25.3.0 (#627)**
- **py-argparse-manpage: add v4.6 (#615)**
- **sentieon-genomics: adding v202503.01 (#610)**
- **Gnu Stow: Add v2.4.1 (#658)**
- **README.md: add header (#649)**
- **py-torchmetrics: add v1.7.3, v1.7.4 (#230)**
- **py-argcomplete: add v3.6.2 (#609)**
- **py-anyio: add v4.9.0 (#608)**
- **py-annexremote: add v1.6.6 (#606)**
- **readline: add 8.3 (#586)**
- **rocksdb: add 10.4.2 (#588)**
- **mochi-thallium: requires C compiler (#648)**
- **shred: new package (#623)**
- **gdbm: fix build with gcc@15 (#557)**
- **strace: add v6.15 (#655)**
- **xorg-server: avoid implicit LibraryList to str conversion warning (#664)**
- **grads: avoid implicit LibraryList to str conversion warning (#665)**
- **Add nanobind v2.8.0 (#661)**
- **gcc: fix `microarchitecture_flags_from_target` failure (#671)**
- **perl: add 5.42.0 and 5.40.2 (#629)**
- **OpenBLAS: normalize test symbols on Windows (#637)**
- **py-numba: fix python dependency bounds (#677)**
- **py-pyqtgraph: variant to choose qt backend (#181)**
- **geopm: Add v3.2 (#266)**
- **Add build dependencies for several packages (#107)**
- **Add py cuda bindings package (#173)**
- **intel_gtpin: add 4.5.0 (#236)**
- **Adding samtools 1.21 (#276)**
- **Re::Solve +asan+ubsan (#392)**
- **VMC: fix deprecated ROOT variant dependency, add version 2-1 (#406)**
- **scorep: add version 9.2, continue adding variants (#580)**
- **scorep: fix name of oneapi compilers in configure (#203)**
- **texinfo: add v7.2 (#575)**
- **spack-labeler.py: maintain non-managed labels (#679)**
- **exodusii: add missing compiler dependencies (#689)**
- **psimd: add dependencies on c and cxx (#680)**
- **py-numpy: respect mkl +ilp64 and threads variants (#182)**
- **tk: needs c compiler to build (#692)**
- **libdrm: version is property of spec (#694)**
- **parquet support needs arrow with parquet and fs support (#696)**
- **netlib-lapack: ensure correct use of "complex" in headers on Windows (#652)**
- **esmf: fix intel compiler name issue (#699)**
- **turn off all hipblaslt dependents (#646)**
- **Fix CMake argument mistake in zfp (#691)**
- **pmix: add v5.0.[6-8], v6.0.0; prrte: add v[2-4]; opempi ~internal-pmix (#241)**
- **spherepack: fix to avoid flags nvhpc doesnt understand (#698)**
- **vgm: bump versions, fix C dependency (#405)**
- **Geant4-VMC: add version 6-7-p1, fix geant4 version dependency (#407)**
- **parallelio: add 2.6.6 checksum (#693)**
- **Geant3: add version 4-4, fix deprecated ROOT variant dependency (#409)**
- **nlcglib depends on fmt, new squashfuse versions (#714)**
- **py-pydot: add 4.0.1 (#733)**
- **Update pmix 5.0.3 checksum (#717)**
- **relax cmake requirement (#711)**
- **unittest_cpp: add c dependency (#721)**
- **netcdf-c: prepare the Autotools builder for version 4.9.3 (#736)**
- **slurm: add v24 and v25 (#734)**
- **python-venv: memoize sysconfig call (#731)**
- **py-scikit-learn: add v1.7.1 (#647)**
- **py-blinker: add v1.9.0 (#750)**
- **perl-xml-parser: add compiler dependency (#718)**
- **perl-clone: add compiler dependency (#719)**
- **libmesh: add missing fortran dependency (#742)**
- **py-torchmetrics: add v1.8.0 (#749)**
- **mariadb-c-client: add 3.4.5 and fix build with gcc@15 (#585)**
- **gmp: fix build with gcc@15 (#576)**
- **ncurses: add 6.5-20250705 (#556)**
- **update hashes after rebase (#732)**
- **rocprofiler-systems: add internal-dyninst variant (#651)**
- **libdwarf: remove url for version and add v2.1.0 (#753)**
- **llvm: add v20.1.7 and v20.1.8 (#745)**
- **just: add a dependency on C (#766)**
- **jujutsu: add latest version (#757)**
- **texlive: add v20250308 (#746)**
- **openblas: Add 0.3.30 (#754)**
- **py-bleach: add 6.2.0 (#737)**
- **JAX: add v0.7.0 (#727)**
- **lammps: add v20250722, v20240829.4, deprecate patch releases (#713)**
- **R: A few package fixes for  gcc@14 (#708)**
- **py-numpy: fix support for C++23 compilers (#701)**
- **netcdf-c: avoid overlinking to C++ library (#700)**
- **exmcutils: add v0.6.4, add maintainer, use autoreconf() (#697)**
- **wxwidgets needs gui for ncvis (#687)**
- **Add required dependencies for xnedit (#686)**
- **Fix env mods (#684)**
- **amrex: add new variant gpu_rdc (#685)**
- **py-calver: add 2025.4.17 (#760)**
- **sccache: add v0.10.0, adjust variants and dependencies (#767)**
- **iwyu: add patch for 0.23 (#759)**
- **meshlab: add v2025.07 (#762)**
- **nalu-wind: add version 2.4.0 (#765)**
- **Adding new versions for slate, blaspp, and lapackpp (#183)**
- **py-certifi: add v2025.7.14 (#761)**
- **qt: add v5.15.17 (#755)**
- **mapl: add v2.58.1 (#642)**
- **py-click: add v8.2.1 (#770)**
- **py-numpy: add v2.3.2 (#774)**
- **miopen-hip and rocblas: add hipblaslt variant (#758)**
- **vmc: depends_on cmake, minimum versions (#741)**
- **Kokkos Kernels: Require Kokkos PIC when building shared library (#728)**
- **Changes for NVIDIA HPC SDK 25.7 (#722)**
- **sherpa: filter_file to fix failing texinfo file (#787)**
- **mpich: add v4.3.1 (#769)**
- **neovim: fix build and update versions (#720)**
- **MFEM: add new version v4.8 (#496)**
- **Miscellaneous Fixes for ROCm Recipes (#653)**
- **tools-sdk: use commit hash instead of main branch (#715)**
- **py-tilelang: new package (#399)**
- **dcm2niix: add 1.0.20250506 (#738)**
- **py-bidskit: add v2025.1.30 (#735)**
- **py-httptools: add v0.6.4 (#780)**
- **Python: overhaul windows installer (#541)**
- **nvbandwidth: add v0.8; fix dependency (#751)**
- **root: add v6.36.02 (#801)**
- **sirius depend on cmake@3.25 (#537)**
- **OpenBLAS: can't build with flang 18 and older (#805)**
- **nvshmem: introduce cmake, add new versions 3 and libfabric variant (#504)**
- **mfem: remove unreachable branch (#776)**
- **armcomputelibrary: remove non-ascii chars from variant description (#784)**
- **py-scikit-build-core: add v0.11.5 (#775)**
- **py-typeguard: added version 4.4.4 (#555)**
- **py-cryptography: add v45.0.5 and update dependency packages (#785)**
- **zoxide: add v0.9.8, add c dependency, and rust bounds (#768)**
- **ADD: Xcrysden package (#706)**
- **scorep: fix `gotcha` `prefix` (#788)**
- **mimalloc: add v1.9.4, v2.2.4, v3.1.5 (#827)**
- **sentencepiece: depends_on("c") (#816)**
- **Add GLVis versions 4.3.0, 4.3.2, and 4.4 (#800)**
- **cppcheck: add v2.18.0 (#822)**
- **node-js: add v24.2.0 (#246)**
- **docs: add modified Spack Packages logo for the packages repository (#660)**
- **cpptrace: v1.0.0 (#261)**
- **mold: add v2.40.2 (#829)**
- **Remove pre-generated versioneer file to fix for python@3.12: (#752)**
- **py-docutils: add v0.21.2 (#798)**
- **py-datalad-container: add v1.2.6 (#836)**
- **py-datalad: add v1.2.1 (#834)**
- **py-cycler: add v0.12.1 (#826)**
- **genie: add v3.6.2 (#813)**
- **py-jinja2: add v3.1.6 (#818)**
- **tests: make independent of spack (#838)**
- **gaudi: add v40.0 (#812)**
- **cublasmp: new package for NVIDIA cuBLASMp (#596)**
- **py_contourpy: Adjust py-setuptools dependency (#518)**
- **gdrcopy: use `cuda_arch` information (#830)**
- **py-flash-attn: add v2.8.1 (#675)**
- **openmpi: add variant to use cray-xpmem name for package configure when cray-xpmem enabled (#357)**

### CODE REVIEW

**Comment by greenc-FNAL** [comment]
Required to resolve GCC build issue:

```console
ssertionError: gcc: Spec.package can only be called on concrete specs
```

fixed with e4fde9058c

**Review by marcmengel — APPROVED** [review]
I'm not even going to try to really review this in detail; It's all upstream changes, lets roll 'em in.
